src/Testcontainers.FirebirdSql/FirebirdSqlBuilder.cs

```diff
@@ -0,0 +1,153 @@
+
+    public const string DefaultSysdbaPassword = "masterkey";
+
+    private const string TestQueryString = "SELECT 1 FROM RDB$DATABASE;";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FirebirdSqlBuilder" /> class.
+    /// </summary>
+    public FirebirdSqlBuilder()
+        : this(new FirebirdSqlConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FirebirdSqlBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    public FirebirdSqlBuilder(FirebirdSqlConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override FirebirdSqlConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Sets the FirebirdSql database.
+    /// </summary>
+    /// <param name="database">The FirebirdSql database.</param>
+    /// <returns>A configured instance of <see cref="FirebirdSqlBuilder" />.</returns>
+    public FirebirdSqlBuilder WithDatabase(string database)
+    {
+        return Merge(DockerResourceConfiguration, new FirebirdSqlConfiguration(database: database))
+            .WithEnvironment("FIREBIRD_DATABASE", database);
+    }
+
+    /// <summary>
+    /// Sets the FirebirdSql username.
+    /// </summary>
+    /// <param name="username">The FirebirdSql username.</param>
+    /// <returns>A configured instance of <see cref="FirebirdSqlBuilder" />.</returns>
+    public FirebirdSqlBuilder WithUsername(string username)
+    {
+        return Merge(DockerResourceConfiguration, new FirebirdSqlConfiguration(username: username))
+            .WithEnvironment("FIREBIRD_USER", "sysdba".Equals(username, StringComparison.OrdinalIgnoreCase) ? string.Empty : username);
+    }
+
+    /// <summary>
+    /// Sets the FirebirdSql password.
+    /// </summary>
+    /// <param name="password">The FirebirdSql password.</param>
+    /// <returns>A configured instance of <see cref="FirebirdSqlBuilder" />.</returns>
+    public FirebirdSqlBuilder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new FirebirdSqlConfiguration(password: password))
+            .WithEnvironment("FIREBIRD_PASSWORD", password)
+            .WithEnvironment("ISC_PASSWORD", password);
+    }
+
+    /// <inheritdoc />
+    public override FirebirdSqlContainer Build()
+    {
+        Validate();
+
+        var compoundWaitStrategy = Wait.ForUnixContainer()
+            .UntilContainerIsHealthy()
+            .AddCustomWaitStrategy(new WaitUntil(DockerResourceConfiguration));
+        return new FirebirdSqlContainer(
+            WithWaitStrategy(compoundWaitStrategy).DockerResourceConfiguration,
+            TestcontainersSettings.Logger);
```

Comment:
Are both wait strategies really necessary?

Reply:
Good point. The custom one alone is not sufficient, which is why I added the healthy check. It may be sufficient by itself

Reply:
I will do some tests to verify if the health check API alone is sufficient. I think the pull request looks good and is ready for merging. WDYT? Thank you for the pull request and your contribution ✌️.

Reply:
I am content with the state it is right now. Thanks you for taking the time to look it over and putting in so much work yourself, too. For me it is fine to merge it.